src/createStaticPackage.js

```diff
@@ -43,19 +86,34 @@ export default function createStaticPackage({ tmpdir, publicFolders }) {
       validateArchive(archive.pointer(), entries);
       const buffer = Buffer.from(data);
       const hash = crypto.createHash('md5').update(buffer).digest('hex');
+
       resolve({ buffer, hash });
     });
     archive.pipe(stream);
 
-    archive.directory(tmpdir, false, { date: FILE_CREATION_DATE });
+    // We can't use archive.directory() here because it is not deterministic.
+    // https://github.com/archiverjs/node-archiver/issues/383#issuecomment-2252938075
+    const tmpdirFiles = getFilesRecursive(tmpdir);
+    for (const file of tmpdirFiles) {
+      archive.file(file, {
+        name: file.slice(tmpdir.length + 1),
+        prefix: '',
```

Comment:
The `false` argument is converted to this under the hood, so I believe this is the equivalent. https://github.com/archiverjs/node-archiver/blob/0830dea0b3798d14d33b454005628958f4611586/lib/core.js#L618-L619